### ISSUE TITLE
The inference speed is very slow

### ISSUE THREAD

**Issue by ycyy** [studio-dots-ai/dots.tts#6]
The inference speed is very slow, with an RTF of more than 2. I am using an L40 GPU, and I would like to know what the benchmark RTF is for model testing.

**Comment by xlians555** [comment]
You can add the `--optimize` flag in current PyTorch version to boost inference speed. Our test results on H800 (voice clone mode, `generate_stream` interface, default inference setting): RTF is roughly 0.6 for Base/Soar and 0.4 for MF. For further speedup, you may cache the reference audio.

**Comment by ukemamaster** [comment]
@xlians555 Is there any example of `generate_stream` ?

**Comment by xlians555** [comment]
```python
  from dots_tts.runtime import DotsTtsRuntime
  import soundfile as sf
  import torch

  runtime = DotsTtsRuntime.from_pretrained(
      "/path/to/dots_tts_model",
      precision="bfloat16",
      optimize=True,
  )

  # generate_stream yields audio chunks (torch.Tensor, shape (1, samples)) as they
  # are produced — useful for low-latency playback or piping to a client.
  stream = runtime.generate_stream(
      text="Hello, this is a streaming speech synthesis test.",
      prompt_audio_path="/path/to/reference.wav",
      prompt_text="The exact transcript of the reference audio.",
      num_steps=10,
      guidance_scale=1.0,
  )

  # Option A: consume chunk-by-chunk (e.g. push to a player / websocket).
  chunks = []
  for chunk in stream:
      chunks.append(chunk.detach().float().cpu())
      # handle_chunk(chunk)  # your real-time consumer here

  # Option B: concatenate and write the full waveform to disk.
  audio = torch.cat(chunks, dim=-1).squeeze().numpy()
  sf.write("output_stream.wav", audio, runtime.sample_rate)
```
@ukemamaster

**Comment by GalenMarek14** [comment]
Do you have plans to make this work in pure C++, like this: https://github.com/pwilkin/openmoss
It would be great, and inference could be much faster.
@xlians555

**Comment by xlians555** [comment]
Thanks for the suggestion, but currently we don't have plans to implement a pure C++ version.

**Comment by Xerophayze** [comment]
I'm also curious about improving the inference speed. I just implemented this in my project tts-story, and the audio quality is quite good, it does seem to have some issues with punctuation once in awhile. But the inference speed on my RTX 3080 TI with 12 gigs of vram seems to be a little slower than some others but faster than others as well. On my particular setup a sentence like the following:

"Audio book produced by John Doe using TTS-Story"  

Took around 19 to 22 seconds to generate after the model load. I've implemented some optimization so it keeps the model loaded between audio chunks which seems to help. But I have chunks of around 250 characters taking about a minute and a half. When I'm working on Long projects I guess it's not that big of a deal but just wanted to see if there's any tips or tricks to maybe help speed up inference.

I've included a screenshot of what my settings are currently for this particular engine. Are there any changes that you might suggest.

<img width="2358" height="1629" alt="Image" src="https://github.com/user-attachments/assets/286b24d0-a705-4047-a2a1-2964289731d8" />

**Comment by xlians555** [comment]
1  Disable text-normalization (which is somewhat heavy) if the texts are simple. The model itself can handle basic TN tasks.
2  Use `generate` interface to avoid running vocoder in every decode step if you don't need streaming outputs.
3  Try shorter reference audio to avoid decoding with large compile buckets if quality is not compromised.
4  You can also try the latest version. I have removed lingua language detection (it wastes ~6 seconds on the first run for language identification) and added prompt cache to avoid repeated computation for identical prompts. 

However, the core inference procedure may still be slow, and we are planning relevant improvements.

**Comment by SerooshSaba** [comment]
How much speedup do you experience with `--optimize` flag?
